### ISSUE TITLE
Unprotected use of process.env. SANITY_INTERNAL_ENV breaks Studio when it's created from client, as in @sanity/astro

### ISSUE THREAD

**Issue by narration-sd** [sanity-io/sanity#6988]
**Describe the bug**

With sanity@3.47.0, Astro studio editors are broken, giving an error that `process` can't be found. 

This is correct, as the code where the crash comes is [here](https://github.com/sanity-io/sanity/blob/df89f884287da39e800117a67fa005560935e7cf/packages/sanity/src/core/studio/packageVersionStatus/checkForLatestVersions.ts#L8), and it is executing _on the client_ in this non-NextJs(!!!) case.

There may be other crash points, as there are plenty of these tests around, none of them protected by assuring `process` exists, which would be an easy fix, explicitly or with `process?.env?.VAR` etc., depending on usage.. 

**To Reproduce**

Create an @sanity/astro project, where you have permission to edit, and try to launch the Studio with its integrated ability. The editor will fail with a message about not finding `process, including when it is side-by-side with a working app screen, in Presentation.

**Expected behavior**

Studio should  'just work'', in all usual ways

**Workaround**

Until this can be fixed, it appears putting a dummy definition in a `vite: `section of `astro.config.ts` will provide enough to make things operate.  The general practice, though, of unprotected accesses to `process`, should be changed. 

```
    define: {
      'process.env.SANITY_INTERNAL_ENV': JSON.stringify(process.env.SANITY_INTERNAL_ENV || 'default_value')
    },
```
 
n..b. Vite config is only allowed in this place as of Astro 4.0; a vite.config.ts will be ignored.

sanity versions
@sanity/cli (global)  3.43.0 (latest: 3.47.0)
@sanity/astro          3.1.3 (up to date)
@sanity/image-url      1.0.2 (up to date)
@sanity/react-loader  1.10.3 (up to date)
@sanity/ui             2.4.0 (up to date)
@sanity/vision        3.47.0 (up to date)
sanity                3.47.0 (up to date)

**What operating system are you using?**

Windows 11 very latest up-to-date

**Which versions of Node.js / npm are you running?**

npm -v && node -v
10.2.4
v20.11.1

**Comment by rexxars** [comment]
Thank you for reporting! This should be fixed in [v3.47.1](https://github.com/sanity-io/sanity/releases/tag/v3.47.1)

**Comment by narration-sd** [comment]
@rexxars  Espen, thank you kindly. Indeed the repair/s work, and I am seeing the app (using an advanced form of @sanity/astro) operate nicely, tested both in local dev and on netlify, without my patch.

Fart response, really appreciated, as this will avoid issues for others especially. 

I won't bother you, about another issue of some importance in visual-editing; maybe making an update on it will bring attention, if they're ready for such there. Quite appreciated how there's been a load :)
